### ISSUE TITLE
[SNES] Possible issue with auto joypad polling

### ISSUE THREAD

**Issue by Max833** [higan-emu/ares#970]
**Describe the bug** / **Steps to reproduce the behavior**:

When you are in a soccer game in "Zenkoku Koukou Soccer 2 (Japan)" and want to pause the game with "start", you will notice that it can't be accessed safely without it closing immediately. Some time ago there was a game that had a very similar issue and was later fixed: 

SpellCraft - Aspects of Valor , https://github.com/bsnes-emu/bsnes/commit/4f7a269ba5d053bb824b5ffa0978799331a5f2d8

**Comment by remutro** [comment]
Note: The commit referenced here: https://github.com/bsnes-emu/bsnes/commit/39c37ec2d1bfcc6b2ee8ae8995fb862eb77c9248
already exists in ares.

**Comment by Max833** [comment]
Yes, bsnes has the exact same issue and Spellcraft is working in both emulators, but something is not 100% right yet.

**Comment by FitzRoyX** [comment]
If you tap it quickly enough it does stay paused, however snes9x you can press and hold and it will stay paused. Can anyone check hardware with a flash cart to verify it's not just bad programming on the game's part?

**Comment by Max833** [comment]
Mesen improved auto-read emulation. There is a new test rom as well. Maybe this is helpful. You can read a lot more about this topic in the "SNES development" discord server.

Mesen changes
https://github.com/SourMesen/Mesen2/commit/efcf0d1523995df31a714391ea53a046fb0b4ba4

Source
https://github.com/undisbeliever/snes-test-roms/blob/master/src/hardware-tests/auto-joy-timing-test.asm

ROM
[auto-joy-timing-test.zip](https://github.com/ares-emulator/ares/files/12371570/auto-joy-timing-test.zip)

**Comment by FitzRoyX** [comment]
The test rom is not pass/fail, so we have no idea what min/max results are on a real snes unless someone here tests it

ares
![ares](https://github.com/ares-emulator/ares/assets/57089001/865a3395-a218-46f1-9c0e-323fdbb41300)

mesen2
![mesen2](https://github.com/ares-emulator/ares/assets/57089001/d5ccbd62-a1c6-4618-812f-690b3496b568)

snes9x
![snes9x](https://github.com/ares-emulator/ares/assets/57089001/5653b2c6-376a-4325-8897-80835d316294)

**Comment by Morilli** [comment]
According to what's been posted on the discord, the expected console values are

|       | cur | min       | max       |
|-------|-----|-----------|-----------|
| start | -   | 002A/002C | 0070      |
| end   | -   | 004B/004C | 009C/009B |

which seems to match mesen2's results.

**Comment by Max833** [comment]
Mesen improved auto-read emulation. There is a new test rom as well. Maybe this is helpful. You can read a lot more about this topic:

//This is based on these test roms: https://github.com/undisbeliever/snes-test-roms/blob/master/src/hardware-tests/
//As well as the CPU schematics here: https://github.com/rgalland/SNES_S-CPU_Schematics/

https://github.com/SourMesen/Mesen2/commit/d1aa957a8052cfa87418908dd46c9cad80d287e8#diff-e6a252584eeb754a8d86aced8463d5b12c223f99d5bf8213d3dcfa78655fa077R64

https://github.com/SourMesen/Mesen2/commit/e4d317635fa9c34a7dd17974d33e2ac442554bf6

And:

https://github.com/SourMesen/Mesen2/commit/6b772003647f02f42787a79a2682362a72a544cd

**Comment by Max833** [comment]
bsnes fixed all auto-joypad polling timing as well. =) Maybe ares can port these changes as well?

https://github.com/bsnes-emu/bsnes/pull/351
https://github.com/bsnes-emu/bsnes/issues/322

**Comment by Max833** [comment]
> bsnes fixed all auto-joypad polling timing as well. =) Maybe ares can port these changes as well?
> 
> [bsnes-emu/bsnes#351](https://github.com/bsnes-emu/bsnes/pull/351) [bsnes-emu/bsnes#322](https://github.com/bsnes-emu/bsnes/issues/322)

@Morilli: Do you think it is possible that you fix these bugs in ares as well? It would be **really** appreciated. Or at least getting permission to relicense the commit to ISC for ares...
